src/languageclient.rs

```diff
@@ -2430,23 +2430,32 @@ impl State {
         let line = ctx.lnum - 1;
         let character = ctx.ccol - 1;
 
-        let result = self.textDocument_completion(
-            &json!({
+        let mut result: Result<Value> = Ok(Value::Null);
+        let completion = self.textDocument_completion(&json!({
                 "buftype": "",
                 "languageId": ctx.filetype,
                 "filename": filename,
                 "line": line,
                 "character": character,
-                "handle": false,
-            }).to_params()?,
-        )?;
-        let result: Option<CompletionResponse> = serde_json::from_value(result)?;
-        let result = result.unwrap_or_else(|| CompletionResponse::Array(vec![]));
-        let is_incomplete = match result {
+                "handle": false}).to_params()?)
+            .unwrap_or_else(|err| {
+                result = Err(err);
+                Value::Null
+            });
+        let completion: CompletionResponse = serde_json::from_value(completion)
+            .unwrap_or_else(|err| {
+                if result.is_ok() {
+                    result = Err(err.into());
```

Comment:
Is this going to reassign result when it is ok?

If answer is no, then the result is reassigned to the second error caused by `from_value(null)` instead of the first error returned by language server.

Reply:
Anyway, I think I got the idea what you're trying to achieve. I can make some refinements some time today.